### PR TITLE
docs: rework about auto-routing

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Rework docs to disable auto-routing by default.
- explain defined routes first, auto-routes last

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/incoming/controllers.rst [hunk 1]
```diff
@@ -9,9 +9,118 @@ Controllers are the heart of your application, as they determine how HTTP reques
...
+
+Helpers
+=======
+
+You can define an array of helper files as a class property. Whenever the controller is loaded
+these helper files will be automatically loaded into memory so that you can use their methods anywhere
+inside the controller:
+
+.. literalinclude:: controllers/016.php
+   :lines: 2-
```
Comment (sfadschm) [inline user_guide_src/source/incoming/controllers.rst:45]: Why did this come back? Was it a rebase?

**Comment by sfadschm** [comment]
Looks rational to me 👍

File: user_guide_src/source/incoming/controllers.rst (hunk 1 above)
Comment (kenjis) [reply user_guide_src/source/incoming/controllers.rst:45]: I've started this work before your `:lines:` deletions.
Today I reworked this file, and pasted the old file into it to check the difference.
And I deleted `:lines:` lines, but it seems I forgot to delete the line.

Comment (kenjis) [reply user_guide_src/source/incoming/controllers.rst]: Fixed and rebased.

Comment (sfadschm) [reply user_guide_src/source/incoming/controllers.rst]: 👍

File: user_guide_src/source/incoming/routing.rst [hunk 2]
```diff
@@ -56,7 +23,7 @@ If you expect a GET request, you use the ``get()`` method:
 
 .. literalinclude:: routing/001.php
 
-A route simply takes the URI path on the left, and maps it to the controller and method on the right,
+A route simply takes the URI path (``/``) on the left, and maps it to the controller and method (``Home::index``) on the right,
```
Comment (samsonasik) [inline user_guide_src/source/incoming/routing.rst:26]: "simply" can be removed? https://jameshfisher.com/2017/02/22/dont-use-simply/

Comment (kenjis) [reply user_guide_src/source/incoming/routing.rst]: @samsonasik Thanks! Done.

**Review by paulbalandan — APPROVED** [review]

**Comment by kenjis** [comment]
@paulbalandan Thank you!

**Pull request merged**